tests/tests.js

```diff
@@ -445,11 +445,11 @@ describe('unicodeSets (v) flag', () => {
 						assert.strictEqual(transpiled, expected);
 					}
 					for (const match of fixture.matches || []) {
-						const transpiledRegex = new RegExp(`^${transpiled}$`, getOutputFlags(flag, options));
+						const transpiledRegex = new RegExp(transpiled, getOutputFlags(flag, options));
 						assert.match(match, transpiledRegex);
 					}
 					for (const nonMatch of fixture.nonMatches || []) {
-						const transpiledRegex = new RegExp(`^${transpiled}$`, getOutputFlags(flag, options));
```

Comment:
Adding `^` and `$` does affect the capturing behaviour, which is why the bug is not caught in previous tests. Here we construct test regex directly from the transpiled regex.